components/Navbar.js

```diff
@@ -3,14 +3,136 @@ import Image from "next/image";
+
+  const NETWORKS = {
+    1: "Ethereum Main Network",
+    3: "Ropsten Tesst Network",
+    4: "Rinkeby Test Network",
+    5: "Goerli Test Network",
+    42: "Kovan Test Network",
+    137: "Polygon Mainnet",
+    56: "BSC Mainnet",
+    10: "Optimism",
+    250: "Fantom Opera",
+  };
+
+  useEffect(() => {
+    setProvider(detectProvider());
+  }, []);
+
+  const onLogin = async provider => {
+    if (provider) {
+      const web3 = new Web3(provider);
+      const accounts = await web3.eth.getAccounts();
+      const chainId = await web3.eth.getChainId();
+      if (accounts.length === 0) {
+        console.log("Please connect to Metamask!");
+      } else if (accounts[0] !== currentAccount) {
+        setCurrentAccount(accounts[0]);
+        setIsConnected(true);
+        setProvider(provider);
+        setWeb3(web3);
+        setChainId(chainId);
+      }
+    } else {
+      setGlobalState("openMMPopup", true);
+    }
+  };
+
+  useEffect(() => {
+    if (provider) {
+      if (provider !== window.ethereum) {
+        console.error(
+          "Not window.ethereum provider. Do you have multiple wallet installed?"
+        );
+      }
+    }
+  }, [provider]);
+
+  useEffect(() => {
+    const handleAccountsChanged = async accounts => {
+      const web3Accounts = await web3.eth.getAccounts();
+      if (accounts.length === 0) {
+        onLogout();
+      } else if (accounts[0] !== currentAccount) {
+        setCurrentAccount(accounts[0]);
+      }
+      setGlobalState("web3account", web3Accounts);
+    };
+    const handleChainChanged = async chainId => {
+      const web3ChainId = await web3.eth.getChainId();
+      setChainId(web3ChainId);
+      setGlobalState("chain", chainId);
+    };
+
+    if (isConnected) {
+      provider.on("accountsChanged", handleAccountsChanged);
+      provider.on("chainChanged", handleChainChanged);
+    }
+
+    return () => {
+      if (isConnected) {
+        provider.removeListener("accountsChanged", handleAccountsChanged);
+        provider.removeListener("chainChanged", handleChainChanged);
+      }
+    };
+  });
+
+  const onLogout = () => {
+    setIsConnected(false);
+  };
+
+  const detectProvider = () => {
+    let provider;
+    if (window.ethereum) {
+      provider = window.ethereum;
+    } else if (window.web3) {
+      provider = window.web3.currentProvider;
+    } else {
+      console.warn("No Ethereum browser detected! Check out Metamask");
+    }
+    return provider;
+  };
+
+  const onLoginHandler = async () => {
+    const provider = detectProvider();
+    if (provider) {
+      setGlobalState("isConnecting", true);
+      await provider.request({
+        method: "eth_requestAccounts",
+      });
+    }
+    setGlobalState("isConnecting", false);
+    onLogin(provider);
+  };
+
+  const getCurrentNetwork = chainId => {
+    return NETWORKS[chainId];
+  };
+
+  setGlobalState("chainId", getCurrentNetwork(chainId));
+
   return (
     <>
```

Comment:
You don't need a fraction `<>` element at this point. Your `<footer>` is already your top level element.